packages/oui-pagination/src/index.spec.js

```diff
@@ -0,0 +1,336 @@
+                expect(clickSpy.calls.count()).toEqual(1);
+            });
+        });
+
+        describe("Page size selector", () => {
+            it("should have default page sizes", () => {
+                const element = TestUtils.compileTemplate(`
+                    <oui-pagination
+                        current-offset="1"
+                        page-size="25"
+                        total-items="1000">
+                    </oui-pagination>
+                `);
+
+                const buttons = getProgress(element).querySelectorAll(".oui-pagination-menu__items-list .oui-pagination-menu__item");
+                expect(buttons.length).toEqual(4);
+                expect(buttons[0].hasAttribute("disabled")).toBeTruthy();
+            });
+
+            it("should change page size", () => {
+                const clickSpy = jasmine.createSpy("clickSpy");
+                const element = TestUtils.compileTemplate(`
+                    <oui-pagination
+                        current-offset="1"
+                        page-size="$ctrl.pageSize"
+                        total-items="1000"
+                        on-change="$ctrl.clickHandler($event)">
+                    </oui-pagination>
+                `, {
+                        pageSize: 25,
+                        clickHandler: clickSpy
+                    });
+
+                const pageSizesButtons = getProgress(element).querySelectorAll(".oui-pagination-menu__items-list .oui-pagination-menu__item");
+                const contextController = element.scope().$ctrl;
+
+                angular.element(pageSizesButtons[1]).triggerHandler("click");
+                expect(clickSpy).toHaveBeenCalledWith(jasmine.objectContaining({ pageSize: 50 }));
+                expect(clickSpy.calls.count()).toEqual(1);
+
+                // Set the new page size (one-way)
+                contextController.pageSize = 50;
+                element.scope().$digest();
+                expect(pageSizesButtons[1].hasAttribute("disabled")).toBeTruthy();
+
+                // Count page buttons
+                const pageButtons = getSelector(element).querySelectorAll(".oui-pagination-menu__items-list .oui-pagination-menu__item");
+                expect(pageButtons.length).toEqual(20);
+            });
+
+            it("should have max page size limiting the page size list (without creating a new value in this list)", () => {
+                const element = TestUtils.compileTemplate(`
+                    <oui-pagination
+                        current-offset="1"
+                        page-size="25"
+                        page-size-max="50"
+                        total-items="1000">
+                    </oui-pagination>
+                `);
+
+                const pageSizesButtons = getProgress(element).querySelectorAll(".oui-pagination-menu__items-list .oui-pagination-menu__item");
+                expect(pageSizesButtons.length).toEqual(2);
```

Comment:
It's not easy to know why it's 2(magic number). expect(pageSizesButtons.length).toEqual(PageSizeMax/pageSize) could be easier to understand

Reply:
Indeed, it's not PageSizeMax/pageSize...
I don't see another solution than write a comment to explain why 2.